### PR TITLE
[Debugger] Introduces the run in terminal feature

### LINKED ISSUE #36561: [Debugger] Accommodate debugging bal programs that take user input
**Description:**
$subject, without the user having to remote debug.

**Describe your problem(s)**
In the current implementation, the user has to remote debug the programs that take an user input, since the debug console doesn't allow the intake of inputs. Through this improvement, the user should be able to debug a ballerina program with user inputs in the general debug mode.

### PR DESCRIPTION
**Pull request by SumayyaZiyad**
Fixes #36561 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [x] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/DebugTestRunner.java [hunk 1]
```diff
@@ -149,6 +151,27 @@ public void initDebugSession(DebugUtils.DebuggeeExecutionKind executionKind) thr
...
+            throws BallerinaTestException {
+        HashMap<String, Object> launchConfigs = new HashMap<>();
+        HashMap<String, Object> extendedCapabilities = new HashMap<>();
+        extendedCapabilities.put("supportsReadOnlyEditors", true);
+        launchConfigs.put("capabilities", extendedCapabilities);
+        launchConfigs.put("terminal", terminalKind);
+        initDebugSession(executionKind, launchConfigs);
+
+        return debugClientConnector.getRequestManager().getDidRunInIntegratedTerminal();
+    }
```
Comment (NipunaRanasinghe) [inline tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/DebugTestRunner.java:172]: Can we reuse this constructor within the other one (`initDebugSession(DebugUtils.DebuggeeExecutionKind executionKind)`) to get rid of the duplicated code? WDYT?

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/runinterminal/ProjectBasedRunInTerminalTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import org.ballerinalang.test.context.BallerinaTestException;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ * Test class to test the runInTerminal feature for project based sources.
+ */
+
```
Comment (NipunaRanasinghe) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/runinterminal/ProjectBasedRunInTerminalTest.java:14]: Extra new line

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/runinterminal/ProjectBasedRunInTerminalTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
+package org.ballerinalang.debugger.test.adapter.runinterminal;
```
Comment (NipunaRanasinghe) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/runinterminal/ProjectBasedRunInTerminalTest.java:1]: Missing license header

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/runinterminal/ProjectBasedRunInTerminalTest.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ * Test class to test the runInTerminal feature for project based sources.
+ */
+
+public class ProjectBasedRunInTerminalTest {
```
Comment (NipunaRanasinghe) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/runinterminal/ProjectBasedRunInTerminalTest.java:15]: Shall we also include a negative test case just to ensure that the debug server is not responding to invalid/unsupported terminal names?

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/runinterminal/SingleFileRunInTerminalTest.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
+
```
Comment (NipunaRanasinghe) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/runinterminal/SingleFileRunInTerminalTest.java:1]: missing license header

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/runinterminal/SingleFileRunInTerminalTest.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+import org.ballerinalang.test.context.BallerinaTestException;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ * Test class to test the runInTerminal feature for single files.
+ */
+
```
Comment (NipunaRanasinghe) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/runinterminal/SingleFileRunInTerminalTest.java:16]: additional new line

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/runinterminal/SingleFileRunInTerminalTest.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+public class SingleFileRunInTerminalTest extends BaseTestCase {
+    DebugTestRunner debugTestRunner;
```
Comment (NipunaRanasinghe) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/runinterminal/SingleFileRunInTerminalTest.java:18]: Shall we also include a negative test case just to ensure that the debug server is not responding to invalid/unsupported terminal names?

File: tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/DebugTestRunner.java (hunk 1 above)
Comment (SumayyaZiyad) [reply tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/DebugTestRunner.java:172]: We can set the terminal kind to an empty string, and pass on the debug session initialization to this, thereby avoiding the code duplication.

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/runinterminal/ProjectBasedRunInTerminalTest.java (hunk 4 above)
Comment (SumayyaZiyad) [reply tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/runinterminal/ProjectBasedRunInTerminalTest.java:31]: +1. Updated.

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/runinterminal/SingleFileRunInTerminalTest.java (hunk 7 above)
Comment (SumayyaZiyad) [reply tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/runinterminal/SingleFileRunInTerminalTest.java:33]: +1. Updated.

File: tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/DebugTestRunner.java (hunk 1 above)
Comment (NipunaRanasinghe) [reply tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/DebugTestRunner.java]: Yeah thats what I meant. +1

**Review by NipunaRanasinghe — COMMENTED** [review]
LGTM

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Pull request merged**